lib/smooth/core/network/Wifi.cpp

```diff
@@ -194,23 +200,54 @@ namespace smooth::core::network
         std::stringstream mac;
 
         uint8_t m[6];
-
-        if (esp_wifi_get_mac(WIFI_IF_STA, m) == ESP_OK)
+        
+        wifi_mode_t mode;
+        if(esp_wifi_get_mode(&mode) == ESP_OK) 
         {
-            for (const auto& v : m)
+            esp_err_t ret;
+            if(mode == WIFI_MODE_STA) 
+                ret = esp_wifi_get_mac(WIFI_IF_STA, m);
+            else
+                ret = esp_wifi_get_mac(WIFI_IF_AP, m);
+
+            if (ret == ESP_OK)
             {
-                if (mac.tellp() > 0)
+                for (const auto& v : m)
                 {
-                    mac << "_";
-                }
+                    if (mac.tellp() > 0)
+                    {
+                        mac << "_";
+                    }
 
-                mac << std::hex << static_cast<int>(v);
+                    mac << std::hex << static_cast<int>(v);
+                }
             }
         }
-
         return mac.str();
     }
 
+    esp_err_t Wifi::get_local_mac_address(uint8_t m[6]) const
+    {
+        wifi_mode_t mode;
+        esp_err_t ret;
+        if((ret = esp_wifi_get_mode(&mode)) == ESP_OK)
```

Comment:
No side effects in if-statements please. Move the assignment out of the if.

Reply:
o.k.